SourceX/DiabloUI/selgame.cpp

```diff
@@ -167,6 +169,69 @@ void selgame_GameSelection_Esc()
 	selgame_endMenu = true;
 }
 
+bool IsSPDifficultyChosen()
+{
+	return selgame_Diff_SP_finished;
+}
+
+bool IsDifficultyAllowed(int value, bool isSP)
+{
+	if (value == 0 || (value == 1 && heroLevel >= 20) || (value == 2 && heroLevel >= 30)) {
+		return true;
+	}
+
+	selgame_Free();
+
+	char *msg = NULL;
+	char *gameType = (isSP ? "single-player" : "multiplayer");
+	char *levelReq = (value == 1 ? "20" : "30");
+	char *diffType = (value == 1 ? "Nightmare" : "Hell");
+
+	char buffer[128];
+	sprintf_s(buffer, "Your character must reach level %s before you can enter a %s game of %s difficulty.", levelReq, gameType, diffType);

```

Comment:
This looks to be windows only and probably why almost all builds fail on this PR

Reply:
Don't worry to much about it though, I'm rebasing it and making some adjustments atm :)